sympy/functions/elementary/complexes.py

```diff
@@ -442,6 +433,18 @@ def _eval_rewrite_as_Abs(self, arg, **kwargs):
     def _eval_simplify(self, **kwargs):
         return self.func(factor_terms(self.args[0]))  # XXX include doit?
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        arg0 = self.args[0]
+        x0 = arg0.subs(x, 0)
+        if x0 != 0:
+            return self.func(x0)
+        if cdir != 0:
+            cdir = arg0.dir(x, cdir)
+        return -S.One if re(cdir) < 0 else S.One
```

Comment:
This is `sign(x).leadterm()`. If cdir is complex then I don't think returning `+-1` is correct:
```python
In [2]: sign(1+I).n()
Out[2]: 0.707106781186548 + 0.707106781186548⋅ⅈ
```

Reply:
hmmm but `limit(sign(I*x + 1), x, 0)` is `1` correct ? (This maps to `sign(x + 1).as_leading_term(x, cdir = I)`)

Reply:
Above it says `if x0 != 0: return ...` so we don't get here for `x + 1`.